### PR TITLE
Weaken `akernel_invs` precondition

### PR DESCRIPTION
**Pull request by michaelmcinerney**
This weakens the precondition of `akernel_invs` as described in VER-1301. This also renames `is_schedulable_bool` to `schedulable`, and introduces the abbreviation `ct_schedulable`.

I will continue on with the remainder of VER-1301, namely proving a Hoare triple for `call_kernel` with postcondition `ct_running s --> ct_schedulable s`

### CODE REVIEW

File: proof/invariant-abstract/KHeap_AI.thy
Comment (corlewis) [inline proof/invariant-abstract/KHeap_AI.thy:2272]: Haha, that's some nice cleanup.

File: proof/invariant-abstract/AInvs.thy
Comment (corlewis) [inline proof/invariant-abstract/AInvs.thy:50]: Feel free to ignore this nitpick, but the postcondition doesn't need the outer brackets

**Review by corlewis — APPROVED** [review]

File: spec/abstract/Schedule_A.thy
Comment (mktnk3) [inline spec/abstract/Schedule_A.thy:215]: It is just a matter of taste, but I kind of prefer something like `ct_sched`...

**Review by mktnk3 — APPROVED** [review]
Nice. I can see that most of the changes are coming from the rename.

To be honest, I am not too comfortable with `schedulable` rename. One thing to consider probably is that the name of this test function is `isSchedulable` in both Haskell and C. And `schedulable` is often used as variable names for the return value.

For the abstract, there is also `is_schedulable_opt`, which was there first, I think, and that is why we have `is_scheduable_bool`. We can maybe settle for renaming `is_schedulable_bool` to `is_schedulable` and remove `is_schedulable_opt`.

**Comment by michaelmcinerney** [comment]
> Nice. I can see that most of the changes are coming from the rename.
> 
> To be honest, I am not too comfortable with `schedulable` rename. One thing to consider probably is that the name of this test function is `isSchedulable` in both Haskell and C. And `schedulable` is often used as variable names for the return value (in Haskell mostly).
> 
> For the abstract, there is also `is_schedulable_opt`, which was there first, I think, and that is why we have `is_scheduable_bool`. We can maybe settle for renaming `is_schedulable_bool` to `is_schedulable` and remove `is_schedulable_opt`.

`is_schedulable` is the name of the monadic function in the abstract spec, so we will have to have something else here. I don't think there were any clashes in Refine due to having `schedulable` and then using `schedulable` for the name of the return value (there were a couple with `ct_schedulable`, which is why I have written `ct_schdble` in a couple places). Seeing that there are so many instances of `schedulable` in AInvs, I think `schedulable` is a good way to go

**Comment by mktnk3** [comment]
ah yes, of course. So, there is that correspondence, and this is for the non-monadic one. But doesn’t that mean that ASpec still has a lot of uses of `schedulable` as return value variable?

**Comment by michaelmcinerney** [comment]
Yeah the ASpec uses `schedulable` as the name of the return value a fair bit, but nothing breaks because of it. Isabelle will just name this `schedulablea` if it appears as a green variable in a proof. Should I use `sched` as the name of the return value instead?

**Comment by mktnk3** [comment]
I for myself wasn’t really feeling the need for introducing this renaming to start with, but I suppose you felt the need so you did this, and it’s up to you how to wrap it up...

**Comment by mktnk3** [comment]
Then again, we have `isSchedulable_bool` in Refine, too...

**Comment by michaelmcinerney** [comment]
Gerwin suggested the renaming in VER-1301, and the names themselves. I do think that `is_schedulable_bool` is a bit cumbersome, especially for something that occurs so frequently.

I forgot about `isSchedulable_bool`. If everyone is happy with `schedulable` and `ct_schedulable`, would `schedulable'` for Refine be reasonable?

Seeing that the `call_kernel` lemmas work is already done, I would prefer to merge this PR as it is now, and then come back and do any renaming and modifications after the `call_kernel` lemmas work is merged. (I am a bit worried that rebasing so many commits would be painful.) Is that OK?

**Comment by mktnk3** [comment]
I’ll leave it up to you to judge

**Comment by lsf37** [comment]
I'd be fine with `schedulable` and `schedulable'`.

**Pull request merged**
